### PR TITLE
[Feature] Login from external Identity Providers

### PR DESCRIPTION
**Pull request by dimitur2204**
Add a call to be able to get a `keycloakAccess` token through a third party provider.
Add a `picture` to the `Person` entity that can be taken from the identity providers.

### CODE REVIEW

**Comment by dimitur2204** [comment]
Tests are not working currently.

**Comment by dimitur2204** [comment]
> Tests are not working currently.

Fixed

File: apps/api/src/auth/auth.service.ts [hunk 1]
```diff
@@ -92,6 +87,51 @@ export class AuthService {
...
+        firstName: userInfo.given_name || '',
+        lastName: userInfo.family_name || '',
+        picture: providerDto.picture,
+        keycloakId: userInfo.sub,
+      },
+      // Store keycloakId to the person with same email
+      update: { keycloakId: userInfo.sub },
+      where: { email: userInfo.email },
+    })
+    return $tokenObs
```
Comment (dimitur2204) [inline apps/api/src/auth/auth.service.ts:117]: There might be a better way to do all these with `rxjs` operators, but I am not the best there. If you have any suggestions on how to do that I am open

**Review by kachar — APPROVED** [review]
Thanks for the heavy-lifting session @dimitur2204 
Works good, let's check it out on staging

**Pull request merged**